### ISSUE TITLE
Update project packaging

### ISSUE THREAD

**Issue by riga** [uhh-cms/columnflow#202]
As the project is evolving towards its beta phase, the packaging should be updated such that it would be no longer required to install it as a submodule of an analysis, but rather a standalone package. This issue is meant to keep track of all things needed for that.

- [ ] Update list of bundled files (including shell scripts, etc)
- [ ] Add proper CLI tools for common tasks (e.g. create_analysis)
- [ ] Define minimum requirements properly in setup.

**Comment by pkausw** [comment]
Thanks for creating this issue! One additional thing to maybe keep in mind: When creating a new analysis, we'd probably want to use the python installation where cf is already installed. This means we'd likely want to skip the part where we install python with `conda` imho.

**Comment by riga** [comment]
I don't think this will fly on standard machines (like lxplus), since we also install gfal2 + bindings via conda that we strictly need.

**Comment by pkausw** [comment]
Well, then the user should take care of that, right? Or we could add the instructions how to setup a correct python version (e.g. with conda) to the documentation. I feel like creating a separate python version from a standalone python package is rather unintuitive - if I used a python module to setup something new, I'd expect that it does so starting from the python environment I started it from.

**Comment by riga** [comment]
Also true. I guess it's a trade-off between convenience and control of the env.

Maybe the setup should optionally skip the installation of conda itself, and in this case just add on top the packages that are strictly needed.
